commitizen/cli.py

```diff
@@ -267,7 +268,11 @@ def main():
 
     # This is for the command required constraint in 2.0
     try:
-        args = parser.parse_args()
+        if sys.argv[1] == "check" and not os.isatty(0):
```

Comment:
@jordanSu Could you please explain a bit on what is`os.isatty(0)` and why it's used? My best guess is that this line causes the error. Also, `sys.argv[1] == "check"` does not guarantee running `cz check` or not. e.g, (`cz -n cz_jira check --commit-msg-file some_file`) . This is why `test_check_jira_multiple_commands` passes while `test_check_conventional_commit_succeeds` does not

Reply:
Sure. So in #200 guihao-liang provided a [possible solution](https://stackoverflow.com/questions/35247817/is-there-a-way-for-a-python-script-to-know-if-it-is-being-piped-to). In this link it mentions the usage of `os.isatty(0)`, which is used to detect if the program is being piped. My implementation is use it to detect if the program is being piped first, then trying to read from stdin.
But I was thinking that reading from stdin causes the error, from error it seems that pytest doesn't let me read from stdin in this case.
For `sys.argv[1] == "check"`, I got it. I found that we actually don't need this one, instead, it can be moved to check.py

Reply:
Yes, I think we should move it to check.py. I'm not sure but we might need to mock `os.isatty` in that case

Reply:
Hello @Lee-W , I've submitted a new PR #267 , which has passed the test. Please help review it.

Reply:
Thanks! I'm reviewing it. Just curious, is there any reason why you create a new PR?

Reply:
Yes. I force pushed to my branch few days ago, but I didn't see new commit appear in this PR, so I create a new PR. (But it seems new commits appear again in this PR now)

Reply:
Hmmm... GitHub sometimes got weird these days. Then, I'll close this one and work on the new one